agora/Agora/Effect/TreasuryWithdrawal.hs

```diff
@@ -141,9 +138,11 @@ instance PTryFrom PData (PAsData PTreasuryWithdrawalDatum)
 
        2. Left over assets should be redirected back to Treasury
 
-     It can be more flexiable over...
+    The output order should be:
+
+      1. Reciever outputs. They should be in the same order as the 'receivers' field of the datum.
```

Comment:
How do we ensure they're in the same order?

Reply:
https://github.com/Liqwid-Labs/agora/blob/e7ad000ac7a9642cc31d9a232c8bb219d75bd77c/agora/Agora/Effect/TreasuryWithdrawal.hs#L233-L271

Reply:
Admittedly it's a bit hard to read😅`treasuryOutputAmount` requires `remainingOutputs`, which then calls `checkReceiverOutputs`.